javaee/impl/src/main/java/org/jboss/forge/addon/javaee/jpa/PersistenceOperationsImpl.java

```diff
@@ -181,6 +186,40 @@ public JavaClassSource newEntity(JavaClassSource javaClass, GenerationType idStr
    @Override
    public JavaClassSource newEntity(JavaClassSource javaClass, GenerationType idStrategy, String tableName,
             String idPropertyName, String versionPropertyName)
+   {
+      FieldSource<JavaClassSource> id = beanOperations.addFieldTo(javaClass, "Long", idPropertyName);
+      id.addAnnotation(Id.class);
+      id.addAnnotation(GeneratedValue.class)
+               .setEnumValue("strategy", idStrategy);
+      id.addAnnotation(Column.class)
+               .setStringValue("name", "id")
+               .setLiteralValue("updatable", "false")
+               .setLiteralValue("nullable", "false");
+
+       return newEntity(javaClass, tableName, id, versionPropertyName);
+   }
+
+   @Override
+   public JavaClassSource newEntityEmbeddedId(JavaClassSource javaClass, String tableName, String idPropertyName,
+           String idPropertyType, String versionPropertyName)
+   {
+       // XXX what if the given class ist not @Embeddable?
```

Comment:
Add the @Embdeddable annotation or throw an Exception?

Reply:
`XXX` as in `TODO` here, maybe I messed sth up. ^^
I wIll thrown an exception, modifying existing classes seems wrong here.